### PR TITLE
Thread safety rubocop plugin

### PR DESCRIPTION
**Pull request by omgitsbillryan**
## Description of change
Adds a thread safety rubocop plugin to our linter.

## Testing done
Run the lint locally using the command to only look for thread safety issues.
```
rubocop -r rubocop-thread_safety --only ThreadSafety,Style/GlobalVars,Style/ClassVars,Style/MutableConstant
```
#### Applies to all PRs

- [ ] Appropriate logging
- [ ] Swagger docs have been updated, if applicable
- [ ] Provide link to originating GitHub issue, or connected to it via ZenHub
- [ ] Does not contain any sensitive information (i.e. PII/credentials/internal URLs/etc., in logging, hardcoded, or in specs)
- [ ] Provide which alerts would indicate a problem with this functionality (if applicable)

### CODE REVIEW

**Comment by johnpaulashenfelter** [comment]
@omgitsbillryan just curious if we've had any problems from thread safety in the past that this would pick up

**Comment by annaswims** [comment]
I like it!  How about committing the results of `rubocop --auto-gen-config` so that we can start checking future PR's right away and address existing problems one-by-one?

File: Gemfile [hunk 1]
```diff
@@ -4,6 +4,8 @@ source 'https://rubygems.org'
 
 ruby '2.4.5'
 
+gem 'rubocop-thread_safety'
```
Comment (jholton) [inline Gemfile]: Should this be moved to the `group :development, :test` block?

**Comment by edmangimelli** [comment]
I'm sad by how little this readme tells me
https://github.com/covermymeds/rubocop-thread_safety

**Comment by omgitsbillryan** [comment]
The jenkins build passed 🎉 

But it looks like code climate is unable to load the plugin 🤔 

```
/usr/local/lib/ruby/site_ruby/2.4.0/rubygems/core_ext/kernel_require.rb:117:in `require': cannot load such file -- rubocop-thread_safety (LoadError)
```

**Comment by annaswims** [comment]
ugh!  So, codeclimate doesn't run the version of rubocop from the gemfile.  It does it's own nonsense and  rubocop-thread_safety isn't in their gemfile https://github.com/codeclimate/codeclimate-rubocop/blob/channel-rubocop-0.71/Gemfile 

I could see ditching codeclimate for this.

**Comment by annaswims** [comment]
different rubocop gem, similar issue https://github.com/codeclimate/codeclimate-rubocop/issues/131

**Comment by omgitsbillryan** [comment]
It might be possible to execute the thread safety cop in a separate Jenkins stage purely on the command line rather than include it as a plugin in our `.rubocop.yml` config. This would give us the best of both worlds - except that thread safety exceptions won't show up in our codeclimate report.

**Comment by johnpaulashenfelter** [comment]
We already run rubocop on CI, so why are we bothering running it again on CodeClimate, especially when it adds the "keep versions in sync" issue. 

Isn't rubocop, for our purposes on the platform, like specs -- pass/fail?

**Pull request merged**